src/js/filter.ts

```diff
@@ -0,0 +1,61 @@
+/* global document, window */
```

Comment:
I didn't make changes to this file - only added types. But the changes are enough that Git thinks it's a new file.